modules/seqimporter/src/main/scala/gem/seqimporter/ConfigReader.scala

```diff
@@ -0,0 +1,279 @@
+            case ((cs,as), s) =>
+              val cs2 = lampToContinuum(s).fold(cs) { _ :: cs }
+              val as2 = lampToArc(s).fold(as) { _ :: as }
+              (cs2, as2)
+          }
+          GcalLamp.unsafeFromConfig(continuum.headOption, arc.strengthR(true): _*)
+        }
+      }
+
+      val Filter = Key.enum[GcalFilter]("filter",
+        "none"         -> GcalFilter.None,
+        "ND1.0"        -> GcalFilter.Nd10,
+        "ND1.6"        -> GcalFilter.Nd16,
+        "ND2.0"        -> GcalFilter.Nd20,
+        "ND3.0"        -> GcalFilter.Nd30,
+        "ND4.0"        -> GcalFilter.Nd40,
+        "ND4-5"        -> GcalFilter.Nd45,
+        "ND5.0"        -> GcalFilter.Nd50,
+        "GMOS balance" -> GcalFilter.Gmos,
+        "HROS balance" -> GcalFilter.Hros,
+        "NIR balance"  -> GcalFilter.Nir
+      )
+
+      val Diffuser = Key.enum[GcalDiffuser]("diffuser",
+        "IR"      -> GcalDiffuser.Ir,
+        "visible" -> GcalDiffuser.Visible
+      )
+
+      val Shutter = Key.enum[GcalShutter]("shutter",
+        "Closed" -> GcalShutter.Closed,
+        "Open"   -> GcalShutter.Open
+      )
+
+      val ExposureTime = Key[Duration]("exposureTime")(Read.durSecs)
+      val Coadds       = Key[Int     ]("coadds"      )(Read.int    )
+    }
+  }
+
+  implicit class ConfigOps3(cm: ConfigMap) {
+    def cget[A](k: System#Key[A]): Option[A] =
+      k.legacyGet(cm).map(k.read)
+
+    def uget[A](k: System#Key[A]): A =
+      cgetOrElse(k, sys.error(s"not found: ${k.path}"))
+
+    def cgetOrElse[A](k: System#Key[A], a: => A): A =
+      cget(k).getOrElse(a)
+  }
+
+  def unsafeStep(m: ConfigMap): Step[(InstrumentConfig, Boolean)] = {
+    val observeType   = m.uget(Legacy.Observe.ObserveType)
+    val instrument    = unsafeInstConfig(m.uget(Legacy.Instrument.Instrument), m)
+
+    val step = observeType match {
+      case "BIAS" =>
+        BiasStep(instrument)
+
+      case "DARK" =>
+        DarkStep(instrument)
+
+      case "OBJECT" | "CAL" =>
+        val p = m.cgetOrElse(Legacy.Telescope.P, OffsetP.Zero)
+        val q = m.cgetOrElse(Legacy.Telescope.Q, OffsetQ.Zero)
+        ScienceStep(instrument, TelescopeConfig(p, q))
+
+      case "ARC" | "FLAT" =>
+        val l = m.uget(Legacy.Calibration.Lamp)
+        val f = m.uget(Legacy.Calibration.Filter)
+        val d = m.uget(Legacy.Calibration.Diffuser)
+        val s = m.uget(Legacy.Calibration.Shutter)
+        val e = m.uget(Legacy.Calibration.ExposureTime)
+        val c = m.uget(Legacy.Calibration.Coadds)
+        GcalStep(instrument, GcalConfig(l, f, d, s, e, c))
+
+      case x =>
+        sys.error(s"Unexpected observeType: $x${m.toList.mkString("\n>  ", "\n  ", "")}")
+    }
+
+    step.strengthR(m.uget(Legacy.Observe.Status))
+  }
+
+  def unsafeInstConfig(i: Instrument, m: ConfigMap): InstrumentConfig =
+    i match {
+      case Instrument.Flamingos2 =>
+        val disperser     = m.uget(Legacy.Instrument.F2.Disperser)
```

Comment:
It's all unsafe since it is converting `String`s.  For example if a `Filter` value shows up in the sequence XML and we haven't added an entry for it in the corresponding map, it will blow up.  When the final ODB import is done someday we won't need any of this anymore.